src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -652,78 +674,57 @@ Result SpatialJoinAlgorithms::BoundingBoxAlgorithm() {
       rtree(bgi::quadratic<16>{}, bgi::indexable<Value>{},
             bgi::equal_to<Value>{}, qec_->getAllocator());
   for (size_t i = 0; i < smallerResult->numRows(); i++) {
-    // get point of row i
-    auto geopoint = getPoint(smallerResult, i, smallerResJoinCol);
-    Box bbox;
-    AnyGeometry geometry;
-    rtreeEntry entry{i, std::nullopt, std::nullopt, std::nullopt};
-
-    if (!geopoint) {
-      if (!getAnyGeometry(smallerResult, i, smallerResJoinCol, geometry)) {
-        // nothing to do. When parsing a point or an area fails, a warning
-        // message gets printed at another place and the point/area just gets
-        // skipped
-        continue;
-      }
-      bbox = boost::apply_visitor(BoundingBoxVisitor(), geometry);
-      entry.geometry = std::move(geometry);
-      entry.boundingBox = bbox;
-
-    } else {
-      entry.geoPoint = geopoint.value();
-      // create a box with a side length of at most 1mm to approximate the point
-      bbox = Box(Point(geopoint.value().getLng(), geopoint.value().getLat()),
-                 Point(geopoint.value().getLng() + 0.00000001,
-                       geopoint.value().getLat() + 0.00000001));
+    // add every box together with the additional information into the rtree
+    std::optional<rtreeEntry> entry =
+        getRtreeEntry(smallerResult, i, smallerResJoinCol);
+    if (!entry) {
+      // nothing to do. When parsing a point or an area fails, a warning
+      // message gets printed at another place and the point/area just gets
+      // skipped
+      continue;
     }
-
-    // add every box together with the row number into the rtree
-    rtree.insert(std::make_pair(std::move(bbox), std::move(entry)));
+    rtree.insert(std::make_pair(entry.value().boundingBox_.value(),
```

Comment:
```suggestion
    rtree.insert(std::pair(entry.value().boundingBox_.value(),
```
technically with the function call, the second argument can be evaluated first, and then you can have a moved out value etc...
(an oddity of the C-standard, the `pair{...}` constructor directly doesn't have this issue.